src/celeritas/optical/surface/SurfacePhysicsUtils.hh

```diff
@@ -40,7 +40,7 @@ is_entering_surface(Real3 const& dir, Real3 const& normal)
 CELER_FORCEINLINE_FUNCTION SurfaceTrackPosition
 next_subsurface_position(SurfaceTrackPosition pos, SubsurfaceDirection dir)
 {
-    return pos + to_signed_offset(dir);
+    return SurfaceTrackPosition{pos.get() + to_signed_offset(dir)};
```

Comment:
Two suggested changes:
- Assert the position is valid to begin with (and not off the end) since `.get` just shows a generic OpaqueId assertion
- Be explicit about the casting from signed to unsigned int (especially since if we use mixed integer lengths, we could end up adding signed int to unsigned short/long and overflowing incorrectly)
 
```suggestion
    CELER_EXPECT(pos);
    return SurfaceTrackPosition{
        pos.unchecked_get()
        + static_cast<SurfaceTrackPosition::size_type>(to_signed_offset(dir))};
```